### DISCUSSION TITLE
TextArea setter

### DISCUSSION THREAD

**Discussion by davetapley** [Textualize/textual#3525] [Ideas]
Am I crazy, or is there no way to set the [`text`](https://textual.textualize.io/widgets/text_area/#textual.widgets._text_area.TextArea.text) on a `TextArea`?

I _assume_ this is by design, and I should just remove and recreate?
If so can I do a PR to note that in the docs. If not then can it be added? 😁

**Comment by davep** [comment]
https://textual.textualize.io/widgets/text_area/#textual.widgets._text_area.TextArea.load_text

**Reply by davetapley** [reply]
Arg, I'm pretty sure I searched that page for every possible synonym of 'load', but not 'load' itself 😆

That said, is there a reason it can't be a setter? Given:

https://github.com/Textualize/textual/blob/ddbc91e31b2e1dc8b4504d05b715e7f954113545/src/textual/widgets/_text_area.py#L927-L930

I think just add:

```py
 @text.setter
    def text(self, value):
        self.load_text(value)
```

Then could:

```py
self.query_one(TextArea).text = "new value"
```

**Reply by davep** [reply]
Paging @darrenburns - was there a reason?

**Reply by darrenburns** [reply]
Seems reasonable- can't think of any reason not to

**Reply by davetapley** [reply]
PR incoming ⌨️
